### PR TITLE
Remove rerankdocs size limitation

### PR DESCRIPTION
**Pull request by mnilsson23**

### CODE REVIEW

**Comment by mnilsson23** [comment]
Hey @alessandrobenedetti, this should remove the restriction you encountered with the rerankdocs size being limited to at least the number of docs you requested returned.

**Comment by alessandrobenedetti** [comment]
Hi @mnilsson23 ,is this actually enough ?
Don't we need also the logic in https://issues.apache.org/jira/browse/SOLR-6323 ?
With the how many ?
I have a draft locally, if necessary I can open a PR.

Cheers

**Comment by mnilsson23** [comment]
@alessandrobenedetti  Pushed the rest of the updates!
Let me know if you see anything up with the changes.  This should mimic what Solr's ReRankQuery does now.

**Comment by alessandrobenedetti** [comment]
I was referring exactly to that !
I did the same locally, so no need to open the PR :) 
I will just check the tests I have locally, I let you know tomorrow!

**Comment by alessandrobenedetti** [comment]
I just checked the tests @mnilsson23 , the only ones I added were : 

- verify the case howMany > ReRankDocs
- verify the case howMany > ReRankDocs paginating
- verify the case when how many == reRankDocs
- verify the case when how many == reRankDocs paginating
- verify the case howMany < Re rank
- verify the case howMany < Re rank paginating

I can contribute those tests, not sure how, maybe I can attach a patch here .
I will sort it out

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java [hunk 1]
```diff
@@ -100,36 +103,70 @@ public int getTotalHits() {
...
-      TopDocs topRerankDocs = reRankRescorer.rescore(searcher,
-          mainDocs, howMany);
-      if (boostedPriority != null) {
-        final SolrRequestInfo info = SolrRequestInfo.getRequestInfo();
+
+      ScoreDoc[] mainScoreDocs = mainDocs.scoreDocs;      
+
+      // Create the array for the reRankScoreDocs.
+      // TODO: Shouldn't this be reRankDocs size?  If so it is broken in 
+      //       Solr's ReRankQuery as well
```
Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java]: I agree, this should be reRankDocs

**Comment by diegoceccarelli** [comment]
LGTM, I'm going to merge, some TODOs that we need to review with Christine: 
   * line 116 LTRCollection
   *  ScoreDoc arrays created at 150 and at 159 can probably be removed reusing the original ScoreDoc

@alessandrobenedetti  if you want to contribute with additional unit tests you are welcome, just open a new PR ;)

**Pull request merged**